### DISCUSSION TITLE
Brunch r87 stable 20201227 - surface pro 4

### DISCUSSION THREAD

**Discussion by smaka510** [sebanc/brunch#750] [Q&A]
Hello,
I have a bunch of old surface pro 4 computers that I installed Chrome OS on using "Brunch r87 stable 20201227". It works BRILLIANTLY!!!  I red in the release notes that "options=ipts" on the kernel command line is needed for the ipts touchscreen to work.

I did the following
1. selected ChromeOS in the grub boot menu and press  "e"
2.. skip down to the line starting with "linux" and add the command to the end
linux ($disk,7)/kernel boot=local noresume noswap log level=7 disablevmx=off cros_secure cros_debug **options=ipts**\
3. press control + x to boot with the newly entered command

OMG EVERYTHING WORKS! Even the touchscreen!!!

My knowledge of Linux is limited... how can I make the change permanent?

**Comment by MohamedAlmahmood** [comment]
Is the touch screen stable of the new kernel? I had to switch back to kernel 4.19 as the touch screen was very unstable. 
Now to make the changes permanent refer to this page: https://github.com/sebanc/brunch/issues/603
If you're dual booting with windows then you can make the changes permanent using grub2win.

**Reply by smaka510** [reply]
I did not try using the touch screen in every possible instance. I just installed a bunch of google apps and tested it with those (mostly games). I also tested to make sure the touch screen would still work coming out of sleep. I am looking to make some chromebooks for a few less than fortunate kids to use.

{a}(temporary)quickly press e when you see grub edit and press f10 to boot.
{b}open terminal(Ctrl+Alt+T) first enter shell then type sudo edit-grub-config

ChromeOS has no Terminal and I cannot do this without root access. Am I failing to understand this?

thank you for your input.

**Reply by MohamedAlmahmood** [reply]
To open the terminal first open the browser, then press (Ctrl+Alt+T). A terminal will show up on the browser. Now type shell and hit enter. Then type sudo edit-grub-config.
Add your kernel parameters. 
Let me know if it works. 
Thanks

**Reply by AmroLouwe** [reply]
Thank you so much, your comment has helped me so much to get surface book into chrome os. Soon will try to set it on the hard drive as dual boot

**Comment by smaka510** [comment]
Friend,
I figured it out... the command I entered was "options-ipts" but should have been "options=ipts". I am so used to powershell that I didnt even notice. EVERYTHING IS WORKING. wifi, touchscreen and all!!! thank you so much!

**Reply by MohamedAlmahmood** [reply]
I'm glad its working

**Comment by Xeroka88** [comment]
Hi, i just try to install Chrome Os on my Surface pro 4 and i did everything like in the instruction... 
1. Linus Mint live
2. runnig the sudo commands
3. and it the end it says... the install complete.
but went i turn in off and back on... nothing happens... it try to boot but nothing happens, it just go to boot, maby is because for some reason in my secure boot says " secure boot enable but custom keys...." ? i disable this... but still, nothing.
can someone help..?? Thanks you
